### PR TITLE
[PowerPC] Add custom lowering for ssubo

### PR DESCRIPTION
**Pull request by maryammo**
This patch is to improve the codegen for ssubo node for i32 in 64-bit mode by custom lowering.

### CODE REVIEW

File: llvm/lib/Target/PowerPC/PPCISelLowering.cpp [hunk 1]
```diff
@@ -198,6 +198,10 @@ PPCTargetLowering::PPCTargetLowering(const PPCTargetMachine &TM,
     }
   }
 
+  if (!Subtarget.hasP10Vector() && isPPC64) {
```
Comment (syzaara) [inline llvm/lib/Target/PowerPC/PPCISelLowering.cpp:201]: nit: don't need the {

**Review by syzaara — APPROVED** [review]
LGTM

File: llvm/lib/Target/PowerPC/PPCISelLowering.cpp [hunk 2]
```diff
@@ -11967,6 +11971,36 @@ SDValue PPCTargetLowering::LowerFP_EXTEND(SDValue Op, SelectionDAG &DAG) const {
...
+                              Xor, DAG.getConstant(-1, dl, MVT::i64));
+
+  SDValue Overflow =
+      DAG.getNode(ISD::SUBE, dl, DAG.getVTList(MVT::i64, MVT::Glue), Xor, Addic,
+                  Addic.getValue(1));
+
+  SDValue OverflowTrunc =
+      DAG.getNode(ISD::TRUNCATE, dl, Op.getNode()->getValueType(1), Overflow);
+  SDValue SubTrunc =
+      (Sub->getValueType(0) != Op.getNode()->getValueType(0))
```
Comment (amy-kwan) [inline llvm/lib/Target/PowerPC/PPCISelLowering.cpp:11998]: Nit: Can probably pull out `Op.getNode()->getValueType(0)`.

**Review by amy-kwan — COMMENTED** [review]
LGTM as well, with some nit comments.

File: llvm/lib/Target/PowerPC/PPCISelLowering.cpp [hunk 3]
```diff
@@ -12038,6 +12073,8 @@ SDValue PPCTargetLowering::LowerOperation(SDValue Op, SelectionDAG &DAG) const {
   case ISD::SETCC:              return LowerSETCC(Op, DAG);
   case ISD::INIT_TRAMPOLINE:    return LowerINIT_TRAMPOLINE(Op, DAG);
   case ISD::ADJUST_TRAMPOLINE:  return LowerADJUST_TRAMPOLINE(Op, DAG);
+  case ISD::SSUBO:
+    return LowerSSUBO(Op, DAG);
```
Comment (lei137) [inline llvm/lib/Target/PowerPC/PPCISelLowering.cpp:12077]: nit: should we just follow the same fmt as the other lines?  I realize it's not clang-format approved...

Comment (maryammo) [reply llvm/lib/Target/PowerPC/PPCISelLowering.cpp:12077]: It is how clang format modified it. (tried it again)

**Pull request merged**